src/Compiler/Reports/PlacementReportManager.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2022 The Foedag team
+
+GPL License
+
+Copyright (c) 2022 The Open-Source FPGA Foundation
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "PlacementReportManager.h"
+
+#include <QFile>
+#include <QRegularExpression>
+#include <QTextStream>
+
+#include "CompilerDefines.h"
+#include "NewProject/ProjectManager/project.h"
+#include "Utils/FileUtils.h"
+
+namespace {
+static constexpr const char *REPORT_NAME{"Report Resource Utilization"};
+static constexpr const char *RESOURCES_SPLIT{"blocks of type:"};
+
+static constexpr const char *BLOCKS_COL{"Blocks"};
+static constexpr const char *NETLIST_COL{"Netlist"};
+static constexpr const char *ARCHITECTURE_COL{"Architecture"};
+
+static const QRegExp FIND_TIMINGS{
+    "Placement estimated (critical|setup).*[0-9].*"};
+static const QRegExp FIND_RESOURCES{"Resource usage.*"};
+}  // namespace
+
+namespace FOEDAG {
+
+std::vector<std::string> PlacementReportManager::getAvailableReportIds() const {
+  return {std::string(REPORT_NAME)};
+}
+
+std::unique_ptr<ITaskReport> PlacementReportManager::createReport(
+    const std::string &reportId) {
+  auto projectPath = Project::Instance()->projectPath().toStdString();
+  auto logPath =
+      std::filesystem::path(projectPath) / std::string(PLACEMENT_LOG);
+
+  if (!FileUtils::FileExists(logPath)) return nullptr;
+
+  auto logFile = QFile(QString::fromStdString(logPath.string()));
```

Comment:
can we avoid mixing QString and std::string? I see here a lot of this

Reply:
+1

Reply:
yeah. I used std::string in ITaskReport class, but files are parsed via Qt. it caused lots of conversions back and forth.
now the interface works on Qt classes and conversions were needless. I also removed them from synthesis parser.